lisp_ease/src/queries/plans.lisp

```diff
@@ -0,0 +1,345 @@
+(in-package :le)
+
+;; variables to store object designators in, for having the robot pick and place two objects at the same time.
+(defvar ?obj-desig1 nil)
+(defvar ?obj-desig2 nil)
+
+
+;;------------------- from tutorials ----------------------
+;;example call: (move-to-object (set-grasp-base-pose (make-poses "?PoseCameraStart")) (set-grasp-look-pose (make-poses "?PoseObjStart")))
+(defun move-to-object (?grasping-base-pose ?grasping-look-pose)
+  "Moves the robot into the position which the human had when interacting with an object. The robot is placed at the spot where the human was standing and is looking at the spot where the object was in Virtual Reality.
+?GRASPING-BASE-POSE: The position for the robot base. Aka, where the human feet were. This transform is being calculated by the set-grasp-base-pose function
+?GRASPING-LOOK-POSE: The position which the object had in Virtual Reality, and where the robot should be looking at. This position is calculated by the grasp-look-pose function.
+RETURNS: Errors or a successfull movement action of the robot."
+  (proj:with-projection-environment pr2-proj::pr2-bullet-projection-environment
+  (cpl:top-level 
+    (cpl:seq
+      (pp-plans::park-arms)
+      ;; move the robot to location
+      (exe:perform (desig:an action
+                            (type going)
+                            (target (desig:a location (pose ?grasping-base-pose)))))
+      ;; move the head to look at location
+      (exe:perform (desig:an action
+                            (type looking)
+                            (target (desig:a location (pose ?grasping-look-pose)))))))))
+
+
+;; ---------------------------------------------------------------------------------------
+;; pick up an object function       ------------------------------------------------------
+(defun pick-up-obj (?type)
+  "Picks up an object of the given type.
+?TYPE: The type of the object that is to be picked up.
+RETURNS: Errors or a successfull movement action of the robot."
+  (let* ((?obj-desig nil)
+         (?arm (get-hand)))
+    (proj:with-projection-environment pr2-proj::pr2-bullet-projection-environment
+      (cpl:top-level
+        (setf ?obj-desig
+              (exe:perform (desig:an action
+                                     (type detecting)
+                                     (object (desig:an object (type ?type))))))
+        (print  (desig:reference
+                 (desig:an action
+                           (type picking-up)
+                           (arm ?arm)
+                           (object ?obj-desig))))
+        (exe:perform 
+         (desig:an action
+                   (type picking-up)
+                   (arm ?arm)
+                   (object ?obj-desig)))))))
+
+
+(defun pick-up-object (?grasping-base-pose ?grasping-look-pose ?type &optional (?second nil))
```

Comment:
Look in ``cram_pr2/cram_pr2_fetch_deliver_plans/src/fetch-deliver-plans.lisp`` for fetching and delivering plans. Think about how to incorporate your parameters into the plans.

Reply:
At some point, these plans will completely go away. The integration should be done on the fetch-deliver-designators.lisp side. Which means that ``(set-grasp-base-pose (make-poses "?PoseCameraStart"))`` has to be called in the file. As that file is in Lisp Prolog, to call Lisp function ``set-grasp-base-pose`` one would need to call ``lisp-fun``.